### PR TITLE
OCM-8441 | test: Expose NAT gateway settings for vpc preparation.

### PR DESCRIPTION
**Pull request by xueli181114**
**What this PR does / why we need it**:
We need to automate negative scenarios to make cluster goes into error.
**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story (OCM-xxxx)*:
Fixes #
[OCM-8441](https://issues.redhat.com//browse/OCM-8441) | test: Expose NAT gateway settings for vpc preparation.
**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Build
- [x] CI
- [ ] Documentation
- [ ] Performance
- [ ] Refactor
- [ ] Style
- [ ] Unit tests
- [ ] Subsystem tests

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

**Comment by xueli181114** [comment]
```
lixue@Xue-Lis-MacBook-Pro rosa-classic % terraform apply -var aws_region=us-west-2 -var name=xueli -var enable_nat_gateway=false -auto-approve
data.aws_availability_zones.available: Reading...
data.aws_availability_zones.available: Read complete after 1s [id=us-west-2]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  + create

Terraform will perform the following actions:

  # module.vpc.aws_default_network_acl.this[0] will be created
  + resource "aws_default_network_acl" "this" {
      + arn                    = (known after apply)
      + default_network_acl_id = (known after apply)
      + id                     = (known after apply)
      + owner_id               = (known after apply)
      + tags                   = {
          + "Name" = "xueli-vpc-default"
        }
      + tags_all               = {
          + "Name" = "xueli-vpc-default"
        }
      + vpc_id                 = (known after apply)

      + egress {
          + action          = "allow"
          + from_port       = 0
          + ipv6_cidr_block = "::/0"
          + protocol        = "-1"
          + rule_no         = 101
          + to_port         = 0
        }
      + egress {
          + action     = "allow"
          + cidr_block = "0.0.0.0/0"
          + from_port  = 0
          + protocol   = "-1"
          + rule_no    = 100
          + to_port    = 0
        }

      + ingress {
          + action          = "allow"
          + from_port       = 0
          + ipv6_cidr_block = "::/0"
          + protocol        = "-1"
          + rule_no         = 101
          + to_port         = 0
        }
      + ingress {
          + action     = "allow"
          + cidr_block = "0.0.0.0/0"
          + from_port  = 0
          + protocol   = "-1"
          + rule_no    = 100
          + to_port    = 0
        }
    }

  # module.vpc.aws_default_route_table.default[0] will be created
  + resource "aws_default_route_table" "default" {
      + arn                    = (known after apply)
      + default_route_table_id = (known after apply)
      + id                     = (known after apply)
      + owner_id               = (known after apply)
      + route                  = (known after apply)
      + tags                   = {
          + "Name" = "xueli-vpc-default"
        }
      + tags_all               = {
          + "Name" = "xueli-vpc-default"
        }
      + vpc_id                 = (known after apply)

      + timeouts {
          + create = "5m"
          + update = "5m"
        }
    }

  # module.vpc.aws_default_security_group.this[0] will be created
  + resource "aws_default_security_group" "this" {
      + arn                    = (known after apply)
      + description            = (known after apply)
      + egress                 = (known after apply)
      + id                     = (known after apply)
      + ingress                = (known after apply)
      + name                   = (known after apply)
      + name_prefix            = (known after apply)
      + owner_id               = (known after apply)
      + revoke_rules_on_delete = false
      + tags                   = {
          + "Name" = "xueli-vpc-default"
        }
      + tags_all               = {
          + "Name" = "xueli-vpc-default"
        }
      + vpc_id                 = (known after apply)
    }

  # module.vpc.aws_internet_gateway.this[0] will be created
  + resource "aws_internet_gateway" "this" {
      + arn      = (known after apply)
      + id       = (known after apply)
      + owner_id = (known after apply)
      + tags     = {
          + "Name" = "xueli-vpc"
        }
      + tags_all = {
          + "Name" = "xueli-vpc"
        }
      + vpc_id   = (known after apply)
    }

  # module.vpc.aws_route.public_internet_gateway[0] will be created
  + resource "aws_route" "public_internet_gateway" {
      + destination_cidr_block = "0.0.0.0/0"
      + gateway_id             = (known after apply)
      + id                     = (known after apply)
      + instance_id            = (known after apply)
      + instance_owner_id      = (known after apply)
      + network_interface_id   = (known after apply)
      + origin                 = (known after apply)
      + route_table_id         = (known after apply)
      + state                  = (known after apply)

      + timeouts {
          + create = "5m"
        }
    }

  # module.vpc.aws_route_table.private[0] will be created
  + resource "aws_route_table" "private" {
      + arn              = (known after apply)
      + id               = (known after apply)
      + owner_id         = (known after apply)
      + propagating_vgws = (known after apply)
      + route            = (known after apply)
      + tags             = {
          + "Name" = "xueli-vpc-private-us-west-2a"
        }
      + tags_all         = {
          + "Name" = "xueli-vpc-private-us-west-2a"
        }
      + vpc_id           = (known after apply)
    }

  # module.vpc.aws_route_table.public[0] will be created
  + resource "aws_route_table" "public" {
      + arn              = (known after apply)
      + id               = (known after apply)
      + owner_id         = (known after apply)
      + propagating_vgws = (known after apply)
      + route            = (known after apply)
      + tags             = {
          + "Name" = "xueli-vpc-public"
        }
      + tags_all         = {
          + "Name" = "xueli-vpc-public"
        }
      + vpc_id           = (known after apply)
    }

  # module.vpc.aws_route_table_association.private[0] will be created
  + resource "aws_route_table_association" "private" {
      + id             = (known after apply)
      + route_table_id = (known after apply)
      + subnet_id      = (known after apply)
    }

  # module.vpc.aws_route_table_association.public[0] will be created
  + resource "aws_route_table_association" "public" {
      + id             = (known after apply)
      + route_table_id = (known after apply)
      + subnet_id      = (known after apply)
    }

  # module.vpc.aws_subnet.private[0] will be created
  + resource "aws_subnet" "private" {
      + arn                                            = (known after apply)
      + assign_ipv6_address_on_creation                = false
      + availability_zone                              = "us-west-2a"
      + availability_zone_id                           = (known after apply)
      + cidr_block                                     = "10.0.1.0/24"
      + enable_dns64                                   = false
      + enable_resource_name_dns_a_record_on_launch    = false
      + enable_resource_name_dns_aaaa_record_on_launch = false
      + id                                             = (known after apply)
      + ipv6_cidr_block_association_id                 = (known after apply)
      + ipv6_native                                    = false
      + map_public_ip_on_launch                        = false
      + owner_id                                       = (known after apply)
      + private_dns_hostname_type_on_launch            = (known after apply)
      + tags                                           = {
          + "Name" = "xueli-vpc-private-us-west-2a"
        }
      + tags_all                                       = {
          + "Name" = "xueli-vpc-private-us-west-2a"
        }
      + vpc_id                                         = (known after apply)
    }

  # module.vpc.aws_subnet.public[0] will be created
  + resource "aws_subnet" "public" {
      + arn                                            = (known after apply)
      + assign_ipv6_address_on_creation                = false
      + availability_zone                              = "us-west-2a"
      + availability_zone_id                           = (known after apply)
      + cidr_block                                     = "10.0.101.0/24"
      + enable_dns64                                   = false
      + enable_resource_name_dns_a_record_on_launch    = false
      + enable_resource_name_dns_aaaa_record_on_launch = false
      + id                                             = (known after apply)
      + ipv6_cidr_block_association_id                 = (known after apply)
      + ipv6_native                                    = false
      + map_public_ip_on_launch                        = true
      + owner_id                                       = (known after apply)
      + private_dns_hostname_type_on_launch            = (known after apply)
      + tags                                           = {
          + "Name" = "xueli-vpc-public-us-west-2a"
        }
      + tags_all                                       = {
          + "Name" = "xueli-vpc-public-us-west-2a"
        }
      + vpc_id                                         = (known after apply)
    }

  # module.vpc.aws_vpc.this[0] will be created
  + resource "aws_vpc" "this" {
      + arn                                  = (known after apply)
      + cidr_block                           = "10.0.0.0/16"
      + default_network_acl_id               = (known after apply)
      + default_route_table_id               = (known after apply)
      + default_security_group_id            = (known after apply)
      + dhcp_options_id                      = (known after apply)
      + enable_dns_hostnames                 = true
      + enable_dns_support                   = true
      + enable_network_address_usage_metrics = (known after apply)
      + id                                   = (known after apply)
      + instance_tenancy                     = "default"
      + ipv6_association_id                  = (known after apply)
      + ipv6_cidr_block                      = (known after apply)
      + ipv6_cidr_block_network_border_group = (known after apply)
      + main_route_table_id                  = (known after apply)
      + owner_id                             = (known after apply)
      + tags                                 = {
          + "Name" = "xueli-vpc"
        }
      + tags_all                             = {
          + "Name" = "xueli-vpc"
        }
    }

Plan: 12 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + azs                    = [
      + "us-west-2a",
    ]
  + cluster-private-subnet = [
      + (known after apply),
    ]
  + cluster-public-subnet  = [
      + (known after apply),
    ]
  + node-private-subnet    = [
      + (known after apply),
    ]
  + vpc-cidr               = "10.0.0.0/16"
  + vpc-id                 = (known after apply)
module.vpc.aws_vpc.this[0]: Creating...
module.vpc.aws_vpc.this[0]: Still creating... [10s elapsed]
module.vpc.aws_vpc.this[0]: Creation complete after 13s [id=vpc-097dff21727c0697f]
module.vpc.aws_internet_gateway.this[0]: Creating...
module.vpc.aws_default_route_table.default[0]: Creating...
module.vpc.aws_route_table.private[0]: Creating...
module.vpc.aws_subnet.private[0]: Creating...
module.vpc.aws_subnet.public[0]: Creating...
module.vpc.aws_default_security_group.this[0]: Creating...
module.vpc.aws_route_table.public[0]: Creating...
module.vpc.aws_default_network_acl.this[0]: Creating...
module.vpc.aws_default_route_table.default[0]: Creation complete after 1s [id=rtb-049827ab038c34557]
module.vpc.aws_route_table.private[0]: Creation complete after 1s [id=rtb-03e01c0de44b6ce5a]
module.vpc.aws_subnet.private[0]: Creation complete after 1s [id=subnet-06c268b9b82e24cfc]
module.vpc.aws_route_table_association.private[0]: Creating...
module.vpc.aws_internet_gateway.this[0]: Creation complete after 1s [id=igw-0c5df54990d98bc30]
module.vpc.aws_route_table.public[0]: Creation complete after 1s [id=rtb-0b17c1d8b764e1254]
module.vpc.aws_route.public_internet_gateway[0]: Creating...
module.vpc.aws_route_table_association.private[0]: Creation complete after 1s [id=rtbassoc-0cd25abe076760c0c]
module.vpc.aws_default_security_group.this[0]: Creation complete after 2s [id=sg-0a14acc57e1a9b061]
module.vpc.aws_default_network_acl.this[0]: Creation complete after 3s [id=acl-04a101deadbd89498]
module.vpc.aws_route.public_internet_gateway[0]: Creation complete after 2s [id=r-rtb-0b17c1d8b764e12541080289494]
module.vpc.aws_subnet.public[0]: Still creating... [10s elapsed]
module.vpc.aws_subnet.public[0]: Creation complete after 12s [id=subnet-0f6504c0c5ee419fc]
module.vpc.aws_route_table_association.public[0]: Creating...
module.vpc.aws_route_table_association.public[0]: Creation complete after 0s [id=rtbassoc-0a62da1c05c7958ee]

Apply complete! Resources: 12 added, 0 changed, 0 destroyed.

Outputs:

azs = tolist([
  "us-west-2a",
])
cluster-private-subnet = [
  "subnet-06c268b9b82e24cfc",
]
cluster-public-subnet = [
  "subnet-0f6504c0c5ee419fc",
]
node-private-subnet = [
  "subnet-06c268b9b82e24cfc",
]
vpc-cidr = "10.0.0.0/16"
vpc-id = "vpc-097dff21727c0697f"
lixue@Xue-Lis-MacBook-Pro rosa-classic % aws ec2 describe-nat-gateways --region us-west-2 --filter Name=vpc-id,Values=vpc-097dff21727c0697f
---------------------
|DescribeNatGateways|
+-------------------+
```

File: tests/tf-manifests/aws/vpc/rosa-classic/variables.tf [hunk 1]
```diff
@@ -49,4 +49,9 @@ variable "aws_shared_credentials_files" {
   type        = list(string)
   default     = null
   description = "File path to the AWS shared credentials file. This file typically used by Shared-VPC cluster."
+}
+variable "enable_nat_gateway" {
+  type        = bool
+  default     = true
+  description = "enable_nat_gateway for private subnets or not. For cluster creation, if enable_nat_gateway=false cluster will be error"
 }
```
Comment (yasun1) [inline tests/tf-manifests/aws/vpc/rosa-classic/variables.tf:57]: Newline required

Comment (xueli181114) [reply tests/tf-manifests/aws/vpc/rosa-classic/variables.tf:60]: Fixed.

**Pull request merged**